Adapters/LdapAdapter.py

```
@@ -0,0 +1,336 @@
+                                resource['perunResourceId'] + ')'
+        resources_string += ')'
+        result_groups = []
+        unique_ids = []
+        groups = self._connector.search_for_entities(
+            self._ldap_base,
+            '(&(uniqueMember=perunUserId=' + user.id + ', ou=People,' +
+            self._ldap_base + ')' + resources_string + ')',
+            ['perunGroupId', 'cn', 'perunUniqueGroupName',
+             'perunVoId', 'uuid', 'description']
+        )
+        for group in groups:
+            if group['perunGroupId'] not in unique_ids:
+                result_groups.append(
+                   self._create_internal_representation_group(group)
+                )
+                unique_ids.append(group['perunGroupId'])
+
+        self._logger.debug('Groups - ' + str(result_groups))
+
+        return result_groups
+
+    def get_facilities_by_attribute_value(self, attribute):
+        raise AdapterSkipException()
+
+    def get_facility_attributes(self, facility, attributes):
+        raise AdapterSkipException()
+
+    def get_user_ext_source(self, ext_source_name, ext_source_login):
+        raise AdapterSkipException()
+
+    def update_user_ext_source_last_access(self, user_ext_source):
+        raise AdapterSkipException()
+
+    def get_user_ext_source_attributes(self, user_ext_source, attr_names):
+        raise AdapterSkipException()
+
+    def set_user_ext_source_attributes(self, user_ext_source, attr_names):
+        raise AdapterSkipException()
+
+    def get_member_status_by_user_and_vo(self, user, vo):
+        group_id = self._connector.search_for_entity(
+            self._ldap_base,
+            '(&(objectClass=perunGroup)(cn=members)(perunVoId=' + vo.id +
+            ')(uniqueMember=perunUserId=' + user.id + ', ou=People,' +
+            self._ldap_base + '))',
+            ['perunGroupId']
+        )
+
+        if not group_id:
+            return MemberStatusEnum.INVALID

```

Instead of returning INVALID throw exception to call another adapter. We will need to get real user status not just INVALID.